Model/System/Message/Notification/MissingApiTokens.php

```diff
@@ -34,62 +34,102 @@
-    public const ADMIN_RESOURCE = 'Nosto_Tagging::system_nosto_account';
-    private Json $result;
-    private NostoHelperAccount $nostoHelperAccount;
-    private NostoOauthBuilder $oauthMetaBuilder;
+    /**
+     * @var NostoHelperScope
+     */
     private NostoHelperScope $nostoHelperScope;
 
     /**
-     * @param Context $context
-     * @param NostoHelperAccount $nostoHelperAccount
-     * @param NostoOauthBuilder $oauthMetaBuilder
+     * @var NostoHelperAccount
+     */
+    private NostoHelperAccount $nostoHelperAccount;
+
+    /**
+     * @var mixed
+     */
+    private $message;
+
+    /**
+     * Messages constructor.
      * @param NostoHelperScope $nostoHelperScope
-     * @param Json $result
+     * @param NostoHelperAccount $nostoHelperAccount
      */
     public function __construct(
-        Context $context,
-        NostoHelperAccount $nostoHelperAccount,
-        NostoOauthBuilder $oauthMetaBuilder,
         NostoHelperScope $nostoHelperScope,
-        Json $result
+        NostoHelperAccount $nostoHelperAccount
     ) {
-        parent::__construct($context);
-
-        $this->nostoHelperAccount = $nostoHelperAccount;
-        $this->oauthMetaBuilder = $oauthMetaBuilder;
-        $this->result = $result;
         $this->nostoHelperScope = $nostoHelperScope;
+        $this->nostoHelperAccount = $nostoHelperAccount;
     }
 
     /**
-     * @return Json
+     * @return Phrase|string
      */
-    public function execute()
+    public function getText()
     {
-        $response = ['success' => false];
+        return __($this->message);
+    }
 
-        $storeId = $this->_request->getParam('store');
-        $store = $this->nostoHelperScope->getStore($storeId);
-        $account = $store !== null ? $this->nostoHelperAccount->findAccount($store) : null;
+    /**
+     * @return string
+     */
+    public function getIdentity()
+    {
+        return sha1('Nosto_MissingApiTokens_Notification');
+    }
+
+    /**
+     * @return bool
+     */
+    public function isDisplayed()
+    {
+        $store = $this->nostoHelperScope->getStore();
+        $account = $this->nostoHelperAccount->findAccount($store);
 
         if ($store !== null && $account !== null) {
-            $metaData = $this->oauthMetaBuilder->build($store, $account);
-            $response['success'] = true;
-            $response['redirect_url'] = OAuthHelper::getAuthorizationUrl($metaData);
+            if ($account->hasMissingTokens()) {
+                $this->buildMessage($account);
+                return true;
+            }
         }
 
-        return $this->result->setData($response);
+        return false;
+    }
+
+    /**
+     * @return int
+     */
+    public function getSeverity()
+    {
+        return MessageInterface::SEVERITY_CRITICAL;
+    }
+
+    /**
+     * Set the value of the message
+     *
+     * @param Account $account
+     * @return void
+     */
+    private function buildMessage($account)
+    {
+        $message = '';
```

Comment:
not needed

Reply:
This is stupid but the reason I added it there is because otherwise phan complains about "Possible slow SQL statament" 🤦 https://github.com/Nosto/nosto-magento2/runs/7610208622?check_suite_focus=true